### PR TITLE
pythonPackages.cvxpy: init at 1.0.25

### PR DESCRIPTION
**Pull request by drewrisinger**
###### Motivation for this change

Add dependency for upcoming qiskit package #78772 @jonringer 
Depends on #78897 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jonringer** [comment]
please fix eval

**Comment by drewrisinger** [comment]
@GrahamcOfBorg eval

File: pkgs/development/python-modules/cvxpy/default.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
+{ lib
+, pythonOlder
+, buildPythonPackage
+, fetchPypi
+, cvxopt
+, ecos
+, multiprocess
+, numpy
+, osqp
```
Comment (jonringer) [inline pkgs/development/python-modules/cvxpy/default.nix]: looks like osqp doesn't exist yet

```
python-modules/cvxpy/default.nix:1:1 called without required argument 'osqp'
```

Comment (drewrisinger) [reply pkgs/development/python-modules/cvxpy/default.nix]: Whoops. #78883

**Comment by FRidh** [comment]
please rebase

**Comment by drewrisinger** [comment]
Rebased, but still waiting on #78897

**Comment by jonringer** [comment]
merged, sorry about that

**Comment by jonringer** [comment]
@GrahamcOfBorg eval

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -1904,6 +1904,8 @@ in {
 
   cvxopt = callPackage ../development/python-modules/cvxopt { };
 
+  cvxpy = callPackage ../development/python-modules/cvxpy { scs = self.scs; };
```
Comment (jonringer) [inline pkgs/top-level/python-packages.nix]: uneeded, default behavior of callPackage
```suggestion
  cvxpy = callPackage ../development/python-modules/cvxpy { };
```

also causing error:

```
error: while querying the derivation named 'python3.7-cvxpy-1.0.25':
while evaluating the attribute 'out.outPath' at /home/jon/.cache/nixpkgs-review/pr-78898/nixpkgs/lib/customisation.nix:151:13:
while evaluating the attribute 'propagatedBuildInputs' of the derivation 'python3.7-cvxpy-1.0.25' at /home/jon/.cache/nixpkgs-review/pr-78898/nixpkgs/pkgs/development/interpreters/python/mk-python-derivation.nix:105:3:
while evaluating 'getOutput' at /home/jon/.cache/nixpkgs-review/pr-78898/nixpkgs/lib/attrsets.nix:464:23, called from undefined position:
while evaluating anonymous function at /home/jon/.cache/nixpkgs-review/pr-78898/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:156:17, called from undefined position:
attribute 'scs' missing, at /home/jon/.cache/nixpkgs-review/pr-78898/nixpkgs/pkgs/top-level/python-packages.nix:1907:67
```

**Comment by drewrisinger** [comment]
@GrahamcOfBorg build python37Packages.cvxpy python38Packages.cvxpy

**Comment by drewrisinger** [comment]
bot builds fail b/c scs package uses mkl (non-free).

**Review by jonringer — APPROVED** [review]
diff LGTM
commit LGTM
has tests :+1: 

```
[9 built, 19 copied (65.8 MiB), 12.9 MiB DL]
https://github.com/NixOS/nixpkgs/pull/78898
2 package built:
python37Packages.cvxpy python38Packages.cvxpy
```

**Pull request merged**